builtin/providers/dnsimple/resource_dnsimple_record.go

```diff
@@ -104,6 +105,11 @@ func resourceDNSimpleRecordRead(d *schema.ResourceData, meta interface{}) error
 
 	resp, err := provider.client.Zones.GetRecord(provider.config.Account, d.Get("domain").(string), recordID)
 	if err != nil {
+		if err != nil && strings.Contains(err.Error(), "404") {
```

Comment:
Is there a reason for repeating `if err != nil`?

Also, this line reminds me we probably have to improve the error reporting in case of 404. Checking against the 404 string is quite error-prone. I can't recall on the top of my head if there is a better way to handle it, I'll check it by tomorrow.